### PR TITLE
[Merged by Bors] - proposals: add processing metrics for major steps

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation
add information about time spent on proposal processing

### CODE REVIEW

File: proposals/metrics/prometheus.go [hunk 1]
```diff
@@ -48,3 +48,94 @@ var NumBlocksInException = metrics.NewHistogramWithBuckets(
 	},
 	prometheus.ExponentialBuckets(1, 2, 8),
 )
+
+var (
+	// ProposalDuration ...
+	ProposalDuration = metrics.NewHistogramWithBuckets(
```
Comment (dshulyak) [inline proposals/metrics/prometheus.go]: i realized after merging a change to vm that using step as a label is very convenient. it allows to query total hist and breakdown using a single query

**Review by dshulyak — APPROVED** [review]

File: proposals/metrics/prometheus.go (hunk 1 above)
Comment (countvonzero) [reply proposals/metrics/prometheus.go]: @dshulyak changed to what you suggested. it's super handy!
Screenshot from 2022-07-19 16-06-57

Comment (countvonzero) [reply proposals/metrics/prometheus.go]: also moved the metrics inside proposal package.

**Comment by countvonzero** [comment]
bors merge

**Comment by countvonzero** [comment]
bors merge

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**